### PR TITLE
sensors: Redefine SENSOR_CHAN_DISTANCE to be in meters

### PR DESCRIPTION
**Pull request by pfalcon**
Most of sensor channels defined by Zephyr use the main, unscaled
SI unit. We also have SENSOR_CHAN_ALTITUDE which is defined in
meters. So, it only makes sense to define distance in meters too

The only driver supporting SENSOR_CHAN_DISTANCE as of now is
vl53l0x.c, which was updated accordingly.

Fixes: #5693

Signed-off-by: Paul Sokolovsky <paul.sokolovsky@linaro.org>

### CODE REVIEW

**Comment by galak** [comment]
@VVESTM can you review

File: drivers/sensor/vl53l0x/vl53l0x.c [hunk 1]
```diff
@@ -83,7 +83,8 @@ static int vl53l0x_channel_get(struct device *dev,
 			val->val1 = 0;
 		}
 	} else {
-		val->val1 = drv_data->RangingMeasurementData.RangeMilliMeter;
+		val->val1 = drv_data->RangingMeasurementData.RangeMilliMeter / 1000;
+		val->val2 = (drv_data->RangingMeasurementData.RangeMilliMeter % 1000) * 1000;
 	}
 	val->val2 = 0;
```
Comment (VVESTM) [inline drivers/sensor/vl53l0x/vl53l0x.c]: Need to move this line in if (chan == SENSOR_CHAN_PROX) condition as val2 is now used. (line 85 for example)

File: samples/sensor/vl53l0x/src/main.c [hunk 2]
```diff
@@ -27,13 +27,13 @@ void main(void)
...
-		printk("prox is %d\n", prox_value.val1);
+		ret = sensor_channel_get(dev, SENSOR_CHAN_PROX, &value);
+		printk("prox is %d\n", value.val1);
 
 		ret = sensor_channel_get(dev,
 					 SENSOR_CHAN_DISTANCE,
-					 &prox_value);
-		printk("distance is %d\n", prox_value.val1);
+					 &value);
+		printk("distance is %.3fm\n", sensor_value_to_double(&value));
```
Comment (VVESTM) [inline samples/sensor/vl53l0x/src/main.c]: This is not working with printk.
A working solution is to align on what is done for hts221 : 
    include  stdio.h instead of misc/printk.h
    use printf instead of printk

**Review by VVESTM — CHANGES_REQUESTED** [review]
Hi, 

This is not working right now. See my comments for the changes to be done.
(decimal value is always null and not displayed)

In addition, as you update vl53l0x.c file, can you replace the link from this line https://github.com/pfalcon/zephyr/blob/a3e9f72b6288e3ba3afab9f7bc95f0c771c6ebfd/drivers/sensor/vl53l0x/vl53l0x.c#L28 by http://www.st.com/en/embedded-software/stsw-img005.html ?

File: drivers/sensor/vl53l0x/vl53l0x.c (hunk 1 above)
Comment (pfalcon) [reply drivers/sensor/vl53l0x/vl53l0x.c]: D'oh, you caught me, I just wanted to push something before a meeting start for CI to churn it ;-). I'll make sure to actually test on disco_l475_iot1

Comment (pfalcon) [reply drivers/sensor/vl53l0x/vl53l0x.c]: Fixed

File: samples/sensor/vl53l0x/src/main.c (hunk 2 above)
Comment (pfalcon) [reply samples/sensor/vl53l0x/src/main.c]: Fixed

**Comment by pfalcon** [comment]
> In addition, as you update vl53l0x.c file, can you replace the link from this line https://github.com/pfalcon/zephyr/blob/a3e9f72b6288e3ba3afab9f7bc95f0c771c6ebfd/drivers/sensor/vl53l0x/vl53l0x.c#L28 by http://www.st.com/en/embedded-software/stsw-img005.html ?

Done.

New version posted, tested on disco_l475_iot1:

~~~
prox is 0
distance is 0.101m
prox is 1
distance is 0.076m
prox is 1
distance is 0.075m
prox is 0
distance is 8.190m
prox is 0
distance is 8.190m
~~~

**Review by tyrantlizard77 — APPROVED** [review]

File: samples/sensor/vl53l0x/src/main.c [hunk 3]
```diff
@@ -27,13 +27,13 @@ void main(void)
 			return;
 		}
 
-		ret = sensor_channel_get(dev, SENSOR_CHAN_PROX, &prox_value);
-		printk("prox is %d\n", prox_value.val1);
+		ret = sensor_channel_get(dev, SENSOR_CHAN_PROX, &value);
+		printk("prox is %d\n", value.val1);
```
Comment (VVESTM) [inline samples/sensor/vl53l0x/src/main.c]: Is it normal to keep printk here ? Wouldn't be better to use printf everywhere ? 
(Not sure what should be the usage of printk in Zephyr...)

**Review by VVESTM — APPROVED** [review]

File: samples/sensor/vl53l0x/src/main.c (hunk 3 above)
Comment (pfalcon) [reply samples/sensor/vl53l0x/src/main.c]: (Some) other samples have it like that too. And generally, the idea of making changes is "make minimal changes required to achieve the needed effect, split unrelated changes to separate commits". So, if there would be idea to "make samples consistently use printf(), not combination of printk/printf", that should be a separate commit (and PR, apparently). I don't think it's a priority for now - it works, and we have bigger things in queue ;-).

> You missed to include stdio.h for printf

Fixed.

Thanks!

Comment (VVESTM) [reply samples/sensor/vl53l0x/src/main.c]: ok, thanks for explanation

**Review by erwango — APPROVED** [review]

File: samples/sensor/vl53l0x/prj.conf [hunk 4]
```diff
@@ -1,3 +1,4 @@
+CONFIG_STDOUT_CONSOLE=y
```
Comment (MaureenHelm) [inline samples/sensor/vl53l0x/prj.conf]: This change should be in a separate patch, it has nothing to do with sensor units.

File: samples/sensor/vl53l0x/src/main.c [hunk 5]
```diff
@@ -7,12 +7,13 @@
 #include <zephyr.h>
 #include <device.h>
 #include <sensor.h>
+#include <stdio.h>
```
Comment (MaureenHelm) [inline samples/sensor/vl53l0x/src/main.c]: This change should be in a separate patch, it has nothing to do with sensor units.

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
The change in sensor units is fine, but please move the config and and stdio changes to a separate commit.

File: samples/sensor/vl53l0x/prj.conf (hunk 4 above)
Comment (pfalcon) [reply samples/sensor/vl53l0x/prj.conf]: It makes sure there're no inconsistencies in the repo, semantic inconsistencies in this case. Like, build won't fail if git bisect will fall inbetween 2 commits you propose, but there can be some erratic results still. E.g., tests could fail. But we don't have tests for that functionality, but if someone try to build the sample, they will get erratic results.

That said, I'm a big fan of separate commits, and will make this change, if only we had overall consistency with such split commits approach by different developers...

**Comment by pfalcon** [comment]
@MaureenHelm : I split off sample changes into a separate commit. Please approve.

**Review by nashif — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]
